### PR TITLE
[MM-68999] Add SchemaVersion to PropertyGroup for group-specific field schema versioning

### PR DESCRIPTION
**Pull request by davidkrauser**
## Summary

Property groups (like `access_control`) contain a set of fields that define things like user attributes, visibility rules, and access control settings. Until now, there was no way to tell whether those field definitions had changed — only a global compatibility version for the whole property system existed.

This adds a `SchemaVersion` field to property groups so that when the fields within a group change (new fields added, types changed, options restructured), that change can be signaled with a version bump. Consumers can read `schema_version` from the group and know whether they need to handle a newer or older shape.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Ticket Link

https://mattermost.atlassian.net/browse/MM-68999

## Changes

- `PropertyGroup` gains a `schema_version` field (integer, returned in API responses)
- A new constant `AccessControlPropertyGroupSchemaVersion` tracks the current version for the `access_control` group, starting at 1
- DB migration 000193 adds the column with a default of 1, so all existing groups are initialized to 1
- `PreSave()` now defaults `SchemaVersion` to 1 for any value `<= 0`, so callers registering builtin groups don't need to set it explicitly — the `content_flagging` group (and any future groups) get a valid schema version automatically
- Store-level tests covering explicit `SchemaVersion` persistence, the default-to-1 path, and immutability on re-registration
- Unit tests for the `PreSave` defaulting behavior (zero and negative values)

## Release Notes

```release-note
NONE
```

## Change Impact: 🟡 Medium

**Regression Risk:** The changes add a new `SchemaVersion` field across the PropertyGroup model, store layer, and database, affecting data persistence and retrieval paths. However, the migration includes a safe `DEFAULT 1` constraint and existing tests do not validate SchemaVersion behavior, leaving potential edge cases uncovered. Notably, the `PreSave()` method does not initialize SchemaVersion to a default value (unlike the Version field), which could lead to inconsistent state if groups are created in-memory without explicitly setting this field.

**QA Recommendation:** Beyond automated test execution, recommend manual verification that: (1) PropertyGroups registered via the API endpoint return the `schema_version` field correctly in responses, (2) accessing existing PropertyGroups after migration returns `schema_version: 1` as expected, and (3) the access_control group is properly registered with `AccessControlPropertyGroupSchemaVersion`. Additionally, validate behavior in access control workflows where PropertyGroup is referenced.

[Review Change Stack](https://app.coderabbit.ai/change-stack/mattermost/mattermost/pull/36747?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: server/channels/app/server.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func NewServer(options ...Option) (*Server, error) {
 
 	// Register builtin property groups before creating hooks that reference them
 	if err = s.propertyService.RegisterBuiltinGroups([]*model.PropertyGroup{
-		{Name: model.AccessControlPropertyGroupName, Version: model.PropertyGroupVersionV2},
+		{Name: model.AccessControlPropertyGroupName, Version: model.PropertyGroupVersionV2, SchemaVersion: model.AccessControlPropertyGroupSchemaVersion},
 		{Name: model.ContentFlaggingGroupName, Version: model.PropertyGroupVersionV1},
 	}); err != nil {
```
[the remark this answers is not in the document]
Comment (davidkrauser) [reply server/channels/app/server.go:277]: done :+1:

**Review by isacikgoz — APPROVED** [review]
LGTM, we can also embed versioning within the attr field of the property field too.

**Review by mgdelacroix — APPROVED** [review]
LTGM! 👍

File: server/channels/db/migrations/postgres/000193_add_property_groups_schema_version.down.sql
Comment (marianunez) [inline server/channels/db/migrations/postgres/000193_add_property_groups_schema_version.down.sql:1]: @davidkrauser it seems that these migration files were not included in the https://github.com/mattermost/mattermost/blob/master/server/channels/db/migrations/migrations.list

My agent picked up on it and added it in this PR https://github.com/mattermost/mattermost/pull/36764 

That PR shouldn't be too delayed but let me know if you prefer to open another one to add it

Comment (davidkrauser) [reply server/channels/db/migrations/postgres/000193_add_property_groups_schema_version.down.sql:1]: @marianunez don't know how I missed that - thank you for addressing. This is targeted for v11.9, so we can wait for your PR to merge.

**Comment by amyblais** [comment]
@davidkrauser Can you make a copy of this document for schema migrations, fill it in, and let me know when done? https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?tab=t.0#heading=h.1ozc9hcotgot

(Note: Ben Cooke added a review migration skill to the AI marketplace that may help here. Here's a guide on how to install the marketplace on claude code https://github.com/mattermost/mattermost-ai-marketplace?tab=readme-ov-file#quick-start. You will want to install the [review-migration](https://github.com/mattermost/mattermost-ai-marketplace/blob/main/plugins/review-migration) plugin.)
